src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1973,34 +1973,70 @@ internal IOperation CreateBoundTupleOperation(BoundTupleExpression boundTupleExp
         {
             var builder = ArrayBuilder<IInterpolatedStringContentOperation>.GetInstance(parts.Length);
             foreach (var part in parts)
             {
-                if (part.Kind == BoundKind.StringInsert)
+                switch (part)
                 {
-                    builder.Add((IInterpolatedStringContentOperation)Create(part));
+                    case BoundStringInsert stringInsert:
+                        Debug.Assert(!stringInsert.IsInterpolatedStringBuilderAppendCall || usesBuilderPattern);
+                        builder.Add((IInterpolatedStringContentOperation)CreateBoundInterpolationOperation(stringInsert));
+                        break;
+                    case BoundLiteral literal:
+                        builder.Add(CreateBoundInterpolatedStringTextOperation(literal));
+                        break;
+                    case BoundCall call:
+                        {
+                            Debug.Assert(usesBuilderPattern);
+                            Debug.Assert(call.Arguments.Length == 1);
+
+                            var currentArg = call.Arguments[0];
+                            addArgument(currentArg);
+                            break;
+                        }
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(part);
                 }
-                else
+            }
+            return builder.ToImmutableAndFree();
+
+            void addArgument(BoundExpression currentArg)
+            {
+                // If the `AppendLiteral` call didn't take a string, there could be conversions. Walk through the conversions until
+                // we find the literal
+                while (currentArg is BoundConversion c)
                 {
-                    builder.Add(CreateBoundInterpolatedStringTextOperation((BoundLiteral)part));
+                    currentArg = c.Operand;
                 }
+
+                builder.Add(CreateBoundInterpolatedStringTextOperation((BoundLiteral)currentArg));
             }
-            return builder.ToImmutableAndFree();
         }
 
         private IInterpolationOperation CreateBoundInterpolationOperation(BoundStringInsert boundStringInsert)
         {
-            IOperation expression = Create(boundStringInsert.Value);
+            // If the string insert is part of a builder pattern, then the actual thing we want to be in the tree is the first argument of
+            // the AppendFormat call. The rest is synthesized and not user-visible
+            // PROTOTYPE(interp-string): We may want to expose this data when the interpolated string is not being used directly as a string
+            var value = (boundStringInsert.IsInterpolatedStringBuilderAppendCall, boundStringInsert.Value) switch
+            {
+                (false, var v) => v,
+                (true, BoundCall call) => call.Arguments[0],
```

Comment:
>call.Arguments[0] [](http://example.com/codeflow?start=42&length=17)

In general this can be a conversion on top of the actual value from the hole. I think we need a more robust way to locate the original node. Alternatively, we should just expose the calls in a new kind of node.

Reply:
I'm going to revert the IOp changes for now and just disable the new tests in the test hook run.